.github/workflows/main.yml

```diff
@@ -0,0 +1,55 @@
+name: ci
+
+on:
+  pull_request:
+    branches:
+      - newsite
+
+  push:
+    branches:
+      - newsite
+
+jobs:
+  test:
+    name: Test run
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os:
+          - macOS-latest
+          - ubuntu-latest
+          - windows-latest
+    steps:
+      - uses: actions/checkout@master
```

Comment:
do we want to specify the `newsite` branch for this? Obviously it will need to be master when we merge all the changes.

Reply:
Yes, that has to be changed when we merge newsite branch to master.

Reply:
+1 for changing it to `actions/checkout@newsite` now

Reply:
As far as I understand, checkout action need the tag of the repository not the branch. So it is basically ` action/checkout@<tags>` .I tried with `newsite` and got this error in log: https://github.com/numpy/numpy.org/pull/46/checks?check_run_id=249019844

Reply:
My understanding is that the the `@` symbol used in an `action` specifies the version *of the action*: https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstepsuses

Reply:
Thanks @WarrenWeckesser for the link. I am wondering why it is working with this`@master` and the base branch for the PR is `newsite`. [Here](https://github.com/actions/checkout) I see that we can use specific branch using 

```
- uses: actions/checkout@master
  with:
    ref: some-branch

```

Reply:
I'm still figuring this out, so all I can say is that `uses: actions/checkout@master`  means you are, in effect, using the development version of the `checkout` action, i.e. the `master`branch of https://github.com/actions/checkout

Reply:
Yeah! It looks like you are correct. Since it is in this format `{owner}/{repo}/{path}@{ref}`. and using `with` option we are customizing the branch, sha, release version. 

So I think basically it is using this PR branch and latest commit to run the action specified.

Reply:
It looks like the default branch (or SHA) that is checked out is explained in https://help.github.com/en/articles/events-that-trigger-workflows#webhook-events.  In particular, see the `GITHUB_SHA` column of the tables for [`pull_request`](https://help.github.com/en/articles/events-that-trigger-workflows#pull-request-event-pull_request) and [`push`](https://help.github.com/en/articles/events-that-trigger-workflows#push-event-push) events.

Also, README.md for the `checkout` action says 

>By default, the branch or tag ref that triggered the workflow will be checked out. If you wish to check out a different branch, specify that using `with.ref`